core/services/ardupilot_manager/mavlink_proxy/MAVLinkRouter.py

```diff
@@ -29,21 +29,20 @@ def convert_endpoint(endpoint: Endpoint) -> str:
                 return f"-t {endpoint.argument}"
             if endpoint.connection_type == EndpointType.TCPClient:
                 return f"-p {endpoint.place}:{endpoint.argument}"
-            return f"{endpoint.place}:{endpoint.argument}"
+            if endpoint.connection_type == EndpointType.UDPServer:
+                return f"-e {endpoint.place}:{endpoint.argument}"
+            raise NotImplementedError(f"Endpoint of type {endpoint.connection_type} not supported on MavlinkRouter.")
 
-        endpoints = " ".join(["--endpoint " + convert_endpoint(endpoint) for endpoint in self.endpoints()])
+        endpoints = " ".join([convert_endpoint(endpoint) for endpoint in self.endpoints()])
 
         if master.connection_type not in [
             EndpointType.UDPServer,
-            EndpointType.UDPClient,
             EndpointType.Serial,
-            EndpointType.TCPServer,
-            EndpointType.TCPClient,
```

Comment:
Master should also be TCP, check sitl.

Reply:
MavlinkRouter says explicitly on the helper that the master endpoints are Serial or UDP. Have you tried it with TCP?